### ISSUE TITLE
Compile-dependency on libtool archives of p4est and sc

### ISSUE THREAD

**Issue by jmark** [DLR-AMR/t8code#496]
The build script of `t8code` looks for `lib/libp4est.la` and `lib/libsc.la` files in order to properly build.

Even then, when given external paths like so: `configure --with-p4est=... --with-sc=...`.

However, this is, according to my sources, not standard (anymore), and many package distributors of libraries delete these
libtool archive files before bundling.

Thus, t8code should be able to compile without these files.

**Comment by jmark** [comment]
@cburstedde I think this might is also be a problem upstream in the build system of p4est.

**Comment by cburstedde** [comment]
We are using libtool in a pretty standard way. Has there been an update in how to invoke it from automake?
If we are doing it as the majority of gnu packages, we should be on the safe side.

If anything is to be done,  I think this should be updated in libsc, p4est and t8code in the same way, by the most recent documentation of libtool.

Btw, the p4est autoconf logic is ahead of t8code in a couple of small details. Would it make sense to align t8code with the most recent p4est automake/libtool setup first and then recheck what needs to be done?

**Comment by cburstedde** [comment]
I've just make installed libsc, then removed the libsc.la file, and configured p4est --with-libsc= to that path. It built fine on my linux machine. How does it work for y'all with the latest prev3-releases of p4est (API 2:0:0)?

**Comment by sloede** [comment]
> How does it work for y'all with the latest prev3-releases of p4est (API 2:0:0)?

I can give it a try, but I am not sure which version of p4est I should use: 2.8? 2.3.5? Or the prev3-develop branch with bootstrapping? I couldn't find a `prev3-releases` branch :-/

**Comment by cburstedde** [comment]
I've made some major version bumps of everything to libtool API version 2:0:0.
The master branch is now current (previously it was tracking years-old backward compatibility code).

New PRs can go against develop for new features, against master for maintenance-related code/doc/meta.

**Comment by sloede** [comment]
Great, thanks for letting us know! I saw in https://github.com/p4est/p4est.github.io/tree/master/release/ that there is now a v2.8.5 and a v2.8.5.5-... release tarball. Am I correct to assume that v2.8.5.5-... is the most up-to-date one?

**Comment by sloede** [comment]
@jmark I tried to build p4est-2.8.5.5..., then delete the .la files, then build t8code against it, but no luck. I think this still needs a fix in the t8code build system, such that the .la files are not needed anymore. Especially, since according to my understanding, the .la files are redundant if there are `.pc` (pkg-config) files available, which p4est does provide.

@cburstedde Thanks a lot for your help, at the moment I don't see anything that can be done on the p4est side 👍

**Comment by sloede** [comment]
> I've updated t8code to the latest p4est and sc on my linux desktop and it builds fine, both with submodules and with --with-p4est and --with-sc referring to outside make install directories. I've only tested the autotools build.

When using `--with-p4est` and `--with-sc` pointing to a separate p4est build, did you remove the `libp4est.la` and `libsc.la` files from `PREFIX/lib` before trying to build t8code? This is where our efforts fail at the moment (we also use `--disable-static`, but I don't think that should matter).

**Comment by cburstedde** [comment]
I can reproduce this problem now with p4est alone. The current autotools build depends on libsc.la, even when using a make install --with-sc. This will be fixed generically for p4est and t8code in one, but I am inclined to do this separately after the currently pending t8code PR. The update to 2.8.5 is at least no worse than the older versions in this respect.

**Comment by sloede** [comment]
I agree, I would make this a separate PR since it's orthogonal to the existing changes. Once this is fixed in p4est/libsc, it would be great if you could make a new release (e.g. 2.8.6) and then ping me - we will wait with updating the Julia wrappers until then.

**Comment by jmark** [comment]
@cburstedde I checked out PR https://github.com/DLR-AMR/t8code/pull/506 and could compile t8code without any `*la` files. Thank you very much for your effort!
